pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatSpec.java

```diff
@@ -47,28 +50,75 @@ public class DateTimeFormatSpec {
   public static final int MIN_FORMAT_TOKENS = 3;
   public static final int MAX_FORMAT_TOKENS = 4;
 
+  public static final int FORMAT_TIMEFORMAT_POSITION_PIPE = 0;
+  public static final int FORMAT_PATTERN_POSITION_PIPE = 1;
+  public static final int FORMAT_UNIT_POSITION_PIPE = 1;
+  public static final int FORMAT_SIZE_POSITION_PIPE = 2;
+  public static final int FORMAT_TIMEZONE_POSITION_PIPE = 2;
+  public static final int MIN_FORMAT_TOKENS_PIPE = 1;
+  public static final int MAX_FORMAT_TOKENS_PIPE = 3;
+
   private final String _format;
   private final int _size;
   private final DateTimeFormatUnitSpec _unitSpec;
   private final DateTimeFormatPatternSpec _patternSpec;
 
   public DateTimeFormatSpec(String format) {
     _format = format;
-    validateFormat(format);
-    String[] formatTokens = StringUtils.split(format, COLON_SEPARATOR, MAX_FORMAT_TOKENS);
-    if (formatTokens.length == MAX_FORMAT_TOKENS) {
-      _patternSpec = new DateTimeFormatPatternSpec(formatTokens[FORMAT_TIMEFORMAT_POSITION],
-          formatTokens[FORMAT_PATTERN_POSITION]);
-    } else {
-      _patternSpec = new DateTimeFormatPatternSpec(formatTokens[FORMAT_TIMEFORMAT_POSITION]);
-    }
-    if (_patternSpec.getTimeFormat() == TimeFormat.TIMESTAMP) {
-      // TIMESTAMP type stores millis since epoch
-      _size = 1;
-      _unitSpec = new DateTimeFormatUnitSpec("MILLISECONDS");
+    if (_format.matches(COLON_REGEX)) {
+      validateFormat(format);
```

Comment:
Not introduced in this PR, but suggest to first do the split, then validate on the split parts to avoid splitting twice. Or even better, perform the validation alone with the value processing to avoid all unnecessary overhead

Reply:
I changed the return type of the validate call to return the tokens after validating. This avoids calling split twice. Is that fine?